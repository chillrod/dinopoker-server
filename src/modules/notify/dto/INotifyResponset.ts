import { WsResponse } from "@nestjs/websockets";

export interface INotifyResponse {
  status: string;
  session: string;
  message: any;
}
