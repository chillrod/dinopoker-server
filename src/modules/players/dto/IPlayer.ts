export type Player = {
  name: string;
  id: string;
  clientID?: string;
  color: string;
  vote: number | string;
};
