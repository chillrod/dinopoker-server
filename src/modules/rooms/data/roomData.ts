import { Player } from "src/modules/players/dto/IPlayer";

interface IRoomData {
  [room: string]: Player[];
}

export class RoomData {
  room: IRoomData;

  constructor() {
    this.room = {};
  }

  roomExistsCheck(room: string) {
    return this.room.hasOwnProperty(room);
  }

  create(room: string) {
    try {
      if (!this.roomExistsCheck(room)) {
        return (this.room[room] = []);
      }

      throw new Error("Room already exists");
    } catch (err) {
      return err.message;
    }
  }

  assignPlayerDataToRoom(room: string, playerData: Player) {
    try {
      if (this.roomExistsCheck(room)) {
        return this.room[room].push(playerData);
      }

      throw new Error("Cant subscribe your data to current room stream");
    } catch (err) {
      return err.message;
    }
  }

  pick(room: string) {
    try {
      if (this.roomExistsCheck(room)) {
        return this.room[room];
      }

      throw new Error("You attempted to return a empty room data");
    } catch (err) {
      return err.message;
    }
  }
}
