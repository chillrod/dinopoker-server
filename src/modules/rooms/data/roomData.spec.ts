import { Player } from "src/modules/players/dto/IPlayer";
import { RoomData } from "./roomData";

describe("Room Data", () => {
  const player: Player = {
    id: "1",
    color: "#FFF",
    name: "Rod",
    vote: 0,
  };

  it("should return the room data array", () => {
    const data = new RoomData();

    expect(data.room).toEqual({});
  });

  it("should create a new room a the given room name", () => {
    const data = new RoomData();

    data.create("if1Room");

    expect(data.room).toMatchSnapshot();
  });

  it("should throw when user tries to create a existing room", () => {
    const data = new RoomData();

    data.create("if1Room");

    expect(data.create("if1Room")).toEqual("Room already exists");
  });

  it("should save a player data in a room array", () => {
    const data = new RoomData();

    data.create("if1Room");

    data.assignPlayerDataToRoom("if1Room", player);

    expect(data.room["if1Room"]).toContain(player);
  });

  it("should delete a room", () => {
    const data = new RoomData();

    data.create("if1Room");
    
    data.create("if2Room");

    data.assignPlayerDataToRoom("if1Room", player);

    data.clear("if1Room");

    expect(data.room).toEqual({ if2Room: [] });
  });

  it("should pick a room content", () => {
    const data = new RoomData();

    data.create("if1Room");

    data.assignPlayerDataToRoom("if1Room", player);

    expect(data.pick("if1Room")).toContain(player);
  });
});
