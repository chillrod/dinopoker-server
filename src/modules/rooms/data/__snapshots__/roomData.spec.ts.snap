// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room Data should create a new room a the given room name 1`] = `
Object {
  "if1Room": Array [],
}
`;
